lib/evmone/instructions_calls.cpp

```diff
@@ -36,7 +36,8 @@ evmc_status_code call(ExecutionState& state) noexcept
     msg.kind = Kind;
     msg.flags = Static ? uint32_t{EVMC_STATIC} : state.msg->flags;
     msg.depth = state.msg->depth + 1;
-    msg.destination = dst;
+    msg.destination = (Kind == EVMC_CALL) ? dst : state.msg->destination;
```

Comment:
Wait. Shouldn't `CALLCODE` also use `dst`? That was one of the broken part about `CALLCODE`?

Reply:
Message recipient in case of `CALLCODE` is the same as in case of `DELEGATECALL`. They differ in message sender.

Reply:
Ok, this is the address where VALUE goes and where storage is modified... This is still confusing after all these years.